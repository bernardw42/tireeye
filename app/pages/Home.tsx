import React from "react";
import { View, Text, TouchableOpacity, SafeAreaView, Image } from "react-native";
import { MaterialCommunityIcons, FontAwesome5 } from "@expo/vector-icons";
import { NavigationProp, ParamListBase } from "@react-navigation/native";

import white from "../../assets/white.png";
import blob1 from "../../assets/blob1.png";
import blob2 from "../../assets/blob2.png";
import blob3 from "../../assets/blob3.png";

interface Props {
  navigation: NavigationProp<ParamListBase>;
}

export default function Home({ navigation }: Props) {
  return (
    <SafeAreaView className="flex-1 bg-[#023535]">
      <Image source={blob1} className="absolute top-[-50px] right-[-110px] transform -translate-y-1/2 w-[300px] h-[400px]" resizeMode="contain"/>
      <Image source={blob2} className="absolute top-[25%] left-[-130px] w-[300px] h-[400px]"  resizeMode="contain"/>
      <Image source={blob3} className="absolute bottom-[-80px] right-[-100px] transform -translate-y-1/2 w-[500px] h-[400px]" resizeMode="contain"/>

    <View className="flex-1 flex justify-center items-center gap-y-[20px]">
      <View className="">
        <Image source={white} className="w-[260px] h-[260px]"></Image>
      </View>
      
      {/* Buttons */}
      <View className="flex flex-row justify-around items-center gap-x-[20px]">
        <TouchableOpacity
          className="bg-white w-[100px] h-[100px] rounded-2xl justify-center items-center"
          onPress={() => navigation.navigate("Scan")}
        >
          <MaterialCommunityIcons name="tire" size={60} color="#023535" />
          <Text className="text-[#023535] font-semibold mt-[5px] text-[12px]">SCAN</Text>
        </TouchableOpacity>
        <TouchableOpacity
          className="bg-white w-[100px] h-[100px] rounded-2xl justify-center items-center"
          onPress={() => navigation.navigate("Notes")}
        >
          <MaterialCommunityIcons name="note-edit" size={60} color="#023535" />
          <Text className="text-[#023535] font-semibold mt-[5px] text-[12px]">ARCHIVE</Text>
        </TouchableOpacity>
      </View>

      <View className="flex flex-row justify-around items-center gap-x-[20px]">
        <TouchableOpacity
          className="bg-white w-[100px] h-[100px] rounded-2xl justify-center items-center"
          onPress={() => navigation.navigate("Market")}
        >
          <FontAwesome5 name="shopping-basket" size={48} color="#023535" />
          <Text className="text-[#023535] font-semibold mt-[5px] text-[12px]">MARKET</Text>
        </TouchableOpacity>
        <TouchableOpacity
          className="bg-white w-[100px] h-[100px] rounded-2xl justify-center items-center"
          onPress={() => navigation.navigate("Workshop")}
        >
          <FontAwesome5 name="tools" size={45} color="#023535" />
          <Text className="text-[#023535] font-semibold mt-[5px] text-[12px]">WORKSHOP</Text>
        </TouchableOpacity>
      </View>

      <TouchableOpacity
          className="bg-white w-[100px] h-[100px] rounded-2xl justify-center items-center"
          onPress={() => navigation.navigate("Profile")}
        >
          <FontAwesome5 name="user-alt" size={45} color="#023535" />
          <Text className="text-[#023535] font-semibold mt-[5px] text-[12px]">PROFILE</Text>
      </TouchableOpacity>

      {/* <TouchableOpacity className="py-3 bg-white w-[300px] rounded-2xl" onPress={()=> navigation.navigate("Welkam")}>
        <Text className="text-xl font-bold text-center text-[#023535]">Welkam</Text>
      </TouchableOpacity> */}

    </View>
    </SafeAreaView>
  );
}


