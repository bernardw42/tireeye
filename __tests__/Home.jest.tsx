// __tests__/App.test.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../app/pages/Home';

describe('<Home />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<Home navigation={undefined} />).toJSON();
    expect(tree && (tree as renderer.ReactTestRendererJSON).children.length).toBe(4);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Home navigation={undefined} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
