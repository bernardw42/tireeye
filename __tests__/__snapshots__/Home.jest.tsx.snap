// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#023535",
      },
    ]
  }
>
  <RCTImageView
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    internal_analyticTag={null}
    resizeMode="contain"
    source={
      [
        {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        },
      ]
    }
    style={
      {
        "height": 400,
        "overflow": "hidden",
        "position": "absolute",
        "right": -110,
        "top": -50,
        "width": 300,
      }
    }
  />
  <RCTImageView
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    internal_analyticTag={null}
    resizeMode="contain"
    source={
      [
        {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        },
      ]
    }
    style={
      {
        "height": 400,
        "left": -130,
        "overflow": "hidden",
        "position": "absolute",
        "top": "25%",
        "width": 300,
      }
    }
  />
  <RCTImageView
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    internal_analyticTag={null}
    resizeMode="contain"
    source={
      [
        {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        },
      ]
    }
    style={
      {
        "bottom": -80,
        "height": 400,
        "overflow": "hidden",
        "position": "absolute",
        "right": -100,
        "width": 500,
      }
    }
  />
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "display": "flex",
        },
        {
          "justifyContent": "center",
        },
        {
          "alignItems": "center",
        },
        {
          "marginTop": -20,
        },
      ]
    }
  >
    <View
      style={
        [
          [
            {
              "marginTop": 20,
            },
          ],
        ]
      }
    >
      <RCTImageView
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        internal_analyticTag={null}
        resizeMode="cover"
        source={
          [
            {
              "__packager_asset": true,
              "height": 1,
              "scale": 1,
              "uri": "file://assets/full/path/to/directory/name.type",
              "width": 1,
            },
          ]
        }
        style={
          {
            "height": 260,
            "overflow": "hidden",
            "width": 260,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-around",
          },
          {
            "alignItems": "center",
          },
          {
            "marginLeft": -20,
          },
          [
            {
              "marginTop": 20,
            },
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 100,
            "justifyContent": "center",
            "marginLeft": 20,
            "opacity": 1,
            "width": 100,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#023535",
                "fontSize": 60,
              },
              undefined,
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󱢖
        </Text>
        <Text
          style={
            [
              {
                "color": "#023535",
              },
              {
                "fontWeight": "600",
              },
              {
                "marginTop": 5,
              },
              {
                "fontSize": 12,
              },
            ]
          }
        >
          SCAN
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 100,
            "justifyContent": "center",
            "marginLeft": 20,
            "opacity": 1,
            "width": 100,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#023535",
                "fontSize": 60,
              },
              undefined,
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󱞁
        </Text>
        <Text
          style={
            [
              {
                "color": "#023535",
              },
              {
                "fontWeight": "600",
              },
              {
                "marginTop": 5,
              },
              {
                "fontSize": 12,
              },
            ]
          }
        >
          ARCHIVE
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-around",
          },
          {
            "alignItems": "center",
          },
          {
            "marginLeft": -20,
          },
          [
            {
              "marginTop": 20,
            },
          ],
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 100,
            "justifyContent": "center",
            "marginLeft": 20,
            "opacity": 1,
            "width": 100,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#023535",
                "fontSize": 48,
              },
              undefined,
              {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#023535",
              },
              {
                "fontWeight": "600",
              },
              {
                "marginTop": 5,
              },
              {
                "fontSize": 12,
              },
            ]
          }
        >
          MARKET
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": 100,
            "justifyContent": "center",
            "marginLeft": 20,
            "opacity": 1,
            "width": 100,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#023535",
                "fontSize": 45,
              },
              undefined,
              {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            [
              {
                "color": "#023535",
              },
              {
                "fontWeight": "600",
              },
              {
                "marginTop": 5,
              },
              {
                "fontSize": 12,
              },
            ]
          }
        >
          WORKSHOP
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "height": 100,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#023535",
              "fontSize": 45,
            },
            undefined,
            {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {
              "fontWeight": "700",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          [
            {
              "color": "#023535",
            },
            {
              "fontWeight": "600",
            },
            {
              "marginTop": 5,
            },
            {
              "fontSize": 12,
            },
          ]
        }
      >
        PROFILE
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
